hack/make-rules/test.sh

```diff
@@ -32,28 +32,26 @@ export KUBE_CACHE_MUTATION_DETECTOR
 KUBE_PANIC_WATCH_DECODE_ERROR="${KUBE_PANIC_WATCH_DECODE_ERROR:-true}"
 export KUBE_PANIC_WATCH_DECODE_ERROR
 
-kube::test::find_dirs() {
+kube::test::find_go_packages() {
   (
     cd "${KUBE_ROOT}"
-    find -L . -not \( \
-        \( \
-          -path './_artifacts/*' \
-          -o -path './_output/*' \
-          -o -path './cmd/kubeadm/test/*' \
-          -o -path './contrib/podex/*' \
-          -o -path './release/*' \
-          -o -path './target/*' \
-          -o -path './test/e2e/e2e_test.go' \
-          -o -path './test/e2e_node/*' \
-          -o -path './test/e2e_kubeadm/*' \
-          -o -path './test/integration/*' \
-          -o -path './third_party/*' \
-          -o -path './staging/*' \
-          -o -path './vendor/*' \
-        \) -prune \
-      \) -name '*_test.go' -print0 | xargs -0n1 dirname | LC_ALL=C sort -u
-
-    find ./staging -name '*_test.go' -not -path '*/test/integration/*' -prune -print0 | xargs -0n1 dirname | LC_ALL=C sort -u
+
+    # Get a list of all the modules in this workspace.
+    local -a workspace_module_patterns
+    kube::util::read-array workspace_module_patterns < <(go work edit -json | jq -r '.Use[].DiskPath + "/..."')
```

Comment:
Is `jq` a tool that is required to be installed when working on Kubernetes?

I don't mind depending on it because it's common enough, I'm just wondering whether this is or should be documented somewhere.

Reply:
https://github.com/search?q=repo%3Akubernetes%2Fkubernetes+kube%3A%3Autil%3A%3Arequire-jq&type=code

Reply:
Then `kube::util::require-jq` is missing here.

When I had uninstalled it and tried it, the shell printed an error and then `go test` was invoked without targets:

```console
$ make test
+++ [0924 08:30:47] Set GOMAXPROCS automatically to 36
hack/make-rules/test.sh: line 41: jq: command not found
!!! [0924 08:30:47] Call tree:
!!! [0924 08:30:47]  1: hack/make-rules/test.sh:157 kube::test::find_go_packages(...)
no Go files in /nvme/gopath/src/k8s.io/kubernetes
!!! [0924 08:30:47] Call tree:
!!! [0924 08:30:47]  1: hack/make-rules/test.sh:157 kube::test::find_go_packages(...)
!!! [0924 08:30:47] Call tree:
!!! [0924 08:30:47]  1: hack/make-rules/test.sh:157 kube::test::find_go_packages(...)
+++ [0924 08:30:47] Running tests without code coverage and with -race
no Go files in /nvme/gopath/src/k8s.io/kubernetes

=== Errors
no Go files in /nvme/gopath/src/k8s.io/kubernetes

DONE 0 tests, 1 error in 0.032s
```

Reply:
good catch

Reply:
Sadly, error handling in bash just really is that bad.

Reply:
I added require-jq - we could do more, like blow up if it finds 0, but followup.

Reply:
macOS Sequoia (15.0, released a week ago) actually ships a system jq ... but it's from 2022.

```console
$ /usr/bin/jq --version
jq-1.6-159-apple-gcff5336-dirty
```

At some point we are going to have fun with the proliferation of jq and no specific version being tested/used.

Reply:
and `go list -json=...` syntax is pretty much undocumented